integration_tests/build.rs

```diff
@@ -0,0 +1,63 @@
+use std::{
+  env,
+  fs::File,
+  io::Write,
+  path::{Path, PathBuf},
+};
+
+type DynResult<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+fn parse_file(path: &Path) -> DynResult<syn::File> {
+  let content = std::fs::read_to_string(path)?;
+  syn::parse_file(&content).map_err(|e| e.into())
+}
+
+fn main() -> DynResult<()> {
+  let out_dir = env::var_os("OUT_DIR").unwrap();
+  let manifest_dir = env::var("CARGO_MANIFEST_DIR")?;
+  let dest_path = PathBuf::from(&out_dir).join("generated_tests.rs");
+  let manifest_path = PathBuf::from(&manifest_dir);
+
+  let mut f = File::create(dest_path)?;
+  let dap_src = manifest_path.parent().unwrap().join("dap").join("src");
+
+  let ast = parse_file(&dap_src.join("responses.rs"))?;
+
+  for item in ast.items {
+    if let syn::Item::Enum(e) = item {
+      if e.ident != "ResponseBody" {
+        continue;
+      }
+      for variant in e.variants {
+        let test_name = format!(
+          "validate_{}_response",
+          variant.ident.to_string().to_lowercase()
+        );
+        writeln!(f, "  #[test]")?;
```

Comment:
Couldn't you use something like to [quote](https://docs.rs/quote/latest/quote/index.html) to generate the source code, instead of using string formatting? That gives us syntax highlighting and makes things a bit more readable, in my opinion.